psycopg/psycopg/pq/pq_ctypes.py

```diff
@@ -47,7 +47,12 @@ def version() -> int:
 
 @impl.PQnoticeReceiver  # type: ignore
 def notice_receiver(arg: c_void_p, result_ptr: impl.PGresult_struct) -> None:
-    pgconn = cast(arg, POINTER(py_object)).contents.value()
+    pgconn = cast(arg, POINTER(py_object)).contents.value
+
+    if sys.implementation.name != "pypy" or callable(pgconn):
+        # On PyPy this doesn't seem to be a weak reference.
+        pgconn = pgconn()
```

Comment:
This was the main thing that was causing me issues when trying to get this up and running with Django.

Reply:
Not a small difference :\

Can we just test `callable()`?